server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectWriteAheadLogs.java

```diff
@@ -301,9 +324,8 @@ private UUID path2uuid(Path path) {
           // There's a reference to a log file, so skip that server's logs
           Set<UUID> idsToIgnore = candidates.remove(dead);
           if (idsToIgnore != null) {
-            for (UUID id : idsToIgnore) {
-              result.remove(id);
-            }
+            result.keySet().removeAll(idsToIgnore);
+            recoveryLogs.keySet().removeAll(idsToIgnore);
```

Comment:
Will removing from the keyset remove it from the underlying map?

Reply:
yes